src/device-state.coffee

```diff
@@ -143,11 +144,94 @@ module.exports = class DeviceState extends EventEmitter
 			applyTargetHealthy = conf.offlineMode or !@applyInProgress or @applications.fetchesInProgress > 0 or cycleTimeWithinInterval
 			return applyTargetHealthy
 
-	normaliseLegacy: =>
+	migrateLegacyApps: (balenaApi) =>
+		console.log('Migrating ids for legacy app...')
+		@db.models('app').select()
+		.then (apps) =>
+			if apps.length == 0
+				console.log('No app to migrate')
+				return
+			app = apps[0]
+			services = JSON.parse(app.services)
```

Comment:
Might be worth adding a specific erorr here pointing to an error with the super database, helping users and support agents, e.g.
```
try
	services = JSON.parse(app.services)
catch
	throw new Error('Applications table in database is corrupted')
```
I'm not super bothered about this, but could be useful. Might be good to add it as a `TypedError` too